### PR TITLE
Yotam and tomer in training

### PR DESCRIPTION
**Pull request by hillelv**
this is not to be merged! this is to make suggestions

### CODE REVIEW

File: src/main/java/training/ModuleConstants.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package training;
+
+import com.ctre.phoenix6.configs.Slot0Configs;
+
+public class ModuleConstants {
+	public static final double ANGULAR_TOLERANCE = 1.8 / 360;
```
Comment (hillelv) [inline src/main/java/training/ModuleConstants.java]: look into the class `Rotation2d` and use it when using angles

File: src/main/java/training/ModuleConstants.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+	public static final Slot0Configs ASS = new Slot0Configs()
```
Comment (hillelv) [inline src/main/java/training/ModuleConstants.java]: consider changing the name to `ANGULAR_MOTOR_SLOT0_CONFIGS` or something

File: src/main/java/training/Robot.java [hunk 3]
```diff
@@ -1,13 +1,19 @@
...
+    private Module module;
 
     public Robot(){
-        // Boot your subsystems...
+        this.module = new Module();
     }
 
     // Add your subsystems getters...
-
+
```
Comment (hillelv) [inline src/main/java/training/Robot.java]: unnecessary space

File: src/main/java/training/TrainingRobotManager.java [hunk 4]
```diff
@@ -1,14 +1,23 @@
...
     private Robot robot;
-
+    private SmartJoystick joystick;
     @Override
     public void trainingInit() {
         this.robot = new Robot();
+        joystick = new SmartJoystick(JoystickPorts.MAIN);
+        joystick.A.whileTrue(new MoveLinearMotor(robot,joystick));
+        joystick.B.whileTrue(new MoveAngularByPosition(robot, joystick, 2));
+        joystick.X.whileTrue(new MoveAngularByPosition(robot, joystick, 1.5));
```
Comment (hillelv) [inline src/main/java/training/TrainingRobotManager.java]: move this to a function

File: src/main/java/training/TrainingRobotManager.java [hunk 5]
```diff
@@ -19,12 +28,12 @@ public void trainingPeriodic() {
     @Override
     public void teleopInit() {
         // schedule your command...
+        
     }
 
 
     @Override
     public void teleopExit() {
-        // cancel your command...
```
Comment (hillelv) [inline src/main/java/training/TrainingRobotManager.java:27]: add this back

File: src/main/java/training/commands/MoveAngularByPosition.java
Comment (hillelv) [inline src/main/java/training/commands/MoveAngularByPosition.java:1]: ctrl + alt + l

File: src/main/java/training/commands/MoveAngularByPosition.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+import utils.joysticks.Axis;
+import utils.joysticks.SmartJoystick;
+
+public class MoveAngularByPosition extends Command {
+	private Module module;
+	private SmartJoystick joystick;
+	private double position;
+	public MoveAngularByPosition(Robot robot, SmartJoystick joystick, double position) {
+		this.module = robot.getModule();
+		addRequirements(module);
```
Comment (hillelv) [inline src/main/java/training/commands/MoveAngularByPosition.java]: move this to the end of the function

File: src/main/java/training/commands/MoveAngularByPosition.java [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+		this.joystick = joystick;
+		this.position = position;
+	}
+	
+	@Override
+	public void initialize() {
+	
+	}
```
Comment (hillelv) [inline src/main/java/training/commands/MoveAngularByPosition.java]: you can just remove this

File: src/main/java/training/commands/MoveAngularByPosition.java [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+	
+	@Override
+	public boolean isFinished() {
+		return Math.abs(module.getAngularPosition() - position) <= ModuleConstants.ANGULAR_TOLERANCE;
+	}
+	
+	@Override
+	public void end(boolean interrupted) {
+		module.stopAngularMotor();
+		System.out.println(module.getAngularPosition());
```
Comment (hillelv) [inline src/main/java/training/commands/MoveAngularByPosition.java]: use the logger (ask me how)

File: src/main/java/training/commands/MoveLinearMotor.java
Comment (hillelv) [inline src/main/java/training/commands/MoveLinearMotor.java:1]: ctrl + alt + l

File: src/main/java/training/commands/MoveLinearMotor.java [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+import training.subsystems.Module;
+import utils.joysticks.Axis;
+import utils.joysticks.SmartJoystick;
+
+public class MoveLinearMotor extends Command {
+	private Module module;
+	private SmartJoystick joystick;
+	public MoveLinearMotor(Robot robot, SmartJoystick joystick){
+		this.module = robot.getModule();
+		addRequirements(module);
```
Comment (hillelv) [inline src/main/java/training/commands/MoveLinearMotor.java]: move to the end of the function

File: src/main/java/training/commands/MoveLinearMotor.java [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+		this.joystick = joystick;
+		
+	}
+	@Override
+	public void initialize() {
+
+	}
```
Comment (hillelv) [inline src/main/java/training/commands/MoveLinearMotor.java]: remove too

File: src/main/java/training/subsystems/Module.java
Comment (hillelv) [inline src/main/java/training/subsystems/Module.java:1]: ctrl alt l

File: src/main/java/training/subsystems/Module.java [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+	}
+	
+	
+	@Override
+	protected void subsystemPeriodic() {
+	
+	}
+	public Module(){
+		this.linearMotor = new TalonFX(0);
+		this.angularMotor = new TalonFX(1);
```
Comment (hillelv) [inline src/main/java/training/subsystems/Module.java]: magic numbers

File: src/main/java/training/subsystems/Module.java [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+		angularMotor.setPosition(0);
+		angularMotor.getConfigurator().apply(new FeedbackConfigs().withSensorToMechanismRatio(150/7));
```
Comment (hillelv) [inline src/main/java/training/subsystems/Module.java]: magic number

File: src/main/java/training/ModuleConstants.java (hunk 1 above)
Comment (tomer-hershman) [reply src/main/java/training/ModuleConstants.java]: consider changing "look into" to more descriptive name ya'ani

**Pull request closed without merging**